chore(protocol): update derivation pipeline doc for the latest protocol implementation (#21057)
taikoxyz/taiko-mono: packages/protocol/docs/Derivation.md

Reviewer: we do not mention `endOfSubmissionWindowTimestamp` as one of the metadata fields for proposal in the initial section. Should we do that or remove it from here since we are actually not using it before we enable permisionless preconfs?

Reviewer: nit: link to the actual struct definition?

The `proposal.sources` array contains `DerivationSource` objects, each with a `blobSlice` field that serves as the primary mechanism for locating and validating proposal metadata. Responsibilities are split as follows:

- **Forced inclusion submitters** publish their `DerivationSourceManifest` blob on L1 directly as blobs.
- **The proposer** gathers every required derivation source—both their own blocks and any outstanding forced inclusions—and publishes a single `ProposalManifest` within their own `DerivationSource` (**appended last**). The inbox contract guarantees that forced inclusions are proposed as they were posted (without metadata being manipulated).

> The manifest data structures are defined as follows:

Reviewer: Maybe we wanna reference that the canonical definitions are in `LibManifest`? Or alternatively we should delete that file if it is not useful, and just keep the definitions here.

### Proposer and `isLowBondProposal` Validation

> The proposer must have sufficient L2 BondManager balance; if `anchorV4` flags `isLowBondProposal = true`, the proposer’s derivation source is replaced with the default manifest.

Reviewer: should we explain a bit more how this is done? i.e. mentioning the specific event?

Also maybe I forgot... but what happens with forced inclusions, they are not replaced with a default manifest right? if so we should call this out and explain how that is determined

#### Forced Inclusion Submission Requirements

> Users submit forced inclusion transactions directly to L1 by posting blob data containing a `DerivationSourceManifest` struct. To ensure valid forced inclusions that pass validation, the following `BlockManifest` fields are ignored and replaced by inherited values from the parent/proposer during metadata application:

Reviewer: nit: we say that the following values are replaced, but then we include the transactions in the list. Maybe something like this is more general?

```suggestion
Users submit forced inclusion transactions directly to L1 by posting blob data containing a `DerivationSourceManifest` struct. To ensure valid forced inclusions that pass validation, the following `BlockManifest`rules are applied:
```
